nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-b17905676460b0ef.arrow\n",
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-533ce78997a9f97c.arrow\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Gi2XjbC69Of5"
+      },
+      "source": [
+        "Before proceeding with the data augmentation, let's have a look into the baseline dataset 😎!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "GD2xmdrg9OLY",
+        "outputId": "57e36e5a-8aba-4223-e65d-2c72ad69c5bf"
+      },
+      "source": [
+        "print(\"Train set\")\n",
+        "print(\"Total samples: {}\\n\".format(len(emotion_train_ds)))\n",
+        "print(\"A random sample\")\n",
+        "print(\"Text: {} \\nLabel: {}\".format(emotion_train_ds['text'][10], idx2label[emotion_train_ds['label'][10]]))\n",
+        "print(\"\\n\")\n",
+        "\n",
+        "print(\"Test set\")\n",
+        "print(\"Total samples: {}\\n\".format(len(emotion_test_ds)))\n",
+        "print(\"A random sample\")\n",
+        "print(\"Text: {} \\nLabel: {}\".format(emotion_test_ds['text'][10], idx2label[emotion_test_ds['label'][10]]))"
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Train set\n",
+            "Total samples: 30\n",
+            "\n",
+            "A random sample\n",
+            "Text: i feel angered and firey \n",
+            "Label: anger\n",
+            "\n",
+            "\n",
+            "Test set\n",
+            "Total samples: 775\n",
+            "\n",
+            "A random sample\n",
+            "Text: im feeling very peaceful about our wedding again now after having \n",
+            "Label: joy\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "xIJPd8P9S9kW"
+      },
+      "source": [
+        "## GPT3Mix pipeline"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Tkx02t5Z3ZVN"
+      },
+      "source": [
+        "We will use [GPT3Mix](https://arxiv.org/abs/2104.08826) model to generate synthetic but hyper-realistic samples from a mixture of real saples utilizing the [GPT-3](https://arxiv.org/abs/2005.14165) language model. Specifically, GPT3Mix takes two real samples from our dataset, embeds these samples in a carefully designed prompt and generates an augmented mixed sample influenced by the sample sentences.\n",
```

Comment:
We don't really use the GPT3Mix model, but rather the idea behind the paper. So I would rephrase it to denote that we are using large LM's, like GPT3 and GPTJ, to produce these samples, and that this idea is based of off the GPT3Mix paper